src/backend/distributed/commands/multi_copy.c

```diff
@@ -1075,23 +1153,26 @@ SendCopyBinaryFooters(CopyOutState copyOutState, List *connectionList)
  * shard.
  */
 static StringInfo
-ConstructCopyStatement(CopyStmt *copyStatement, int64 shardId)
+ConstructCopyStatement(CopyStmt *copyStatement, int64 shardId, bool useBinaryCopyFormat)
 {
 	StringInfo command = makeStringInfo();
 
-	char *schemaName = copyStatement->relation->schemaname;
-	char *relationName = copyStatement->relation->relname;
-
-	char *shardName = pstrdup(relationName);
+	char *shardSchemaName = copyStatement->relation->schemaname;
+	char *shardTableName = pstrdup(copyStatement->relation->relname);
 	char *shardQualifiedName = NULL;
 
-	AppendShardIdToName(&shardName, shardId);
+	AppendShardIdToName(&shardTableName, shardId);
+	shardQualifiedName = quote_qualified_identifier(shardSchemaName, shardTableName);
 
-	shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
-
-	appendStringInfo(command,
-					 "COPY %s FROM STDIN WITH (FORMAT BINARY)",
-					 shardQualifiedName);
+	if (useBinaryCopyFormat)
+	{
+		appendStringInfo(command, "COPY %s FROM STDIN WITH (FORMAT BINARY)",
+						 shardQualifiedName);
+	}
+	else
+	{
+		appendStringInfo(command, "COPY %s FROM STDIN", shardQualifiedName);
+	}
```

Comment:
Wouldn't it be better to only have one appendStringInfo() here, but store the format in another variable? I.e. something like
```
const char *copyFormat = NULL;
...
if (useBinaryCopyFormat)
{
     copyFormat = "BINARY"; 
}
else
{
     copyFormat = "TEXT"; 
}
```
appendStringInfo(command, "COPY %s FROM STDIN (FORMAT %s)", shardQualifiedName, copyFormat);